department-of-veterans-affairs/caseflow
# Justin/appeals 4534 hide hearing sections

Resolves [APPEALS-4534](https://vajira.max.gov/browse/APPEALS-4534)

### Description
- Updated the Convert to Virtual Hearing form for VSO users to be able to view only certain content that a VSO is allowed to see and not everything that hearings coordinator roles can see
- Wrote feature tests to verify everything runs as expected

### Acceptance Criteria

1. The following sections on the Convert to Virtual form (hearings application) are hidden for VSO users
- [x] Hearing Date
- [x] Hearing Time section
- [x] Relation to Veteran (Appellant section if appellant is not Veteran)
- [x] Veterans Law Judge (VLJ) section

- Mailing Address section 
- [ ] If claimant is Veteran than it reads: Veteran Mailing Address
- [ ] If claimant is not Veteran then it reads Appellant Mailing Address

2. There are no changes to the section visibility defined in A/C 1 for users who are not VSO users

3. The following informational text under the Veteran/Appellant email field is modified for VSO users
- [ ] Current form text: "This email address will be used to send notifications for this hearing only"
- [ ] VSO text: "An incorrect email address will impact the ability to access the hearing"

4. The following informational text continues to appear under the Veteran/Appellant email field for non-VSO users
- [ ] "This email address will be used to send notifications for this hearing only" 
- [ ] no impacts to current form text

### Testing Plan
1. Log in as an intake user
2. Create an appeal where claimant is other than veteran. Pick Bob, who is the spouse of a fake veteran. The appeal with get a POA assigned as paralyzed veterans of America.
3. Set the appeal with a hearing type and add issues. Schedule the appeal.
4. NOTE: Copy down the UUID for ease of use in finding the appeal for testing

5. Log in as hearings coordinator role
6. Go to CASEFLOW Queue and enter queue/appeals/{ uuid }/ in the address bar to find the appeal
7. Schedule the veteran for a central, video, or travel hearing.

8. Log in as VSO user role
9. Route to the link /hearings/{ uuid }/details
10. Observe that the "Convert to virtual hearing" form immediately pulls up, and observe that hearing date, hearing time, relation to veteran, and VLJ sections are hidden.
11. Observe the VSO Text under appellant email section reads "An incorrect email address will impact the ability to access the hearing"


### User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue - shown below
 - [ ] Functionality remains for Hearings users but changes for new VSO user functionality

 BEFORE|AFTER
| Hearings Coordinator View |
![image](https://user-images.githubusercontent.com/95875751/174349249-f4547458-bb05-4570-97f1-c67fd525848b.png)
| VSO User View |
![image](https://user-images.githubusercontent.com/95875751/174349329-bc75c881-07c4-44a1-8668-2d95e4a94956.png)

| Hearings Coordinator View |
![image](https://user-images.githubusercontent.com/95875751/174349357-9adc9bd1-d43b-44a2-bf11-1329207fb0f6.png) 
| VSO User View |
![image](https://user-images.githubusercontent.com/95875751/174349388-c33d5fe1-2e90-4ef9-b57d-fb9439468dd8.png)


### Code Documentation Updates
